apps/system/test/unit/statusbar_test.js

```diff
@@ -702,28 +702,59 @@ suite('system/Statusbar', function() {
   });
 
   suite('lockscreen support', function() {
-    var lockscreenApp, app;
+    var lockscreenApp, app, cloneStatusbarStub;
 
-    setup(function() {
-      lockscreenApp = getApp(false, true);
-      app = getApp(false, false);
+    function lockScreen() {
       MockService.mockQueryWith('getTopMostWindow', lockscreenApp);
       var evt = new CustomEvent('hierarchytopmostwindowchanged', {
         detail: lockscreenApp
       });
       Statusbar.handleEvent(evt);
-    });
+    }
 
-    teardown(function() {
+    function unlockScreen() {
       var evt = new CustomEvent('hierarchytopmostwindowchanged', {
         detail: app
       });
       Statusbar.handleEvent(evt);
+    }
+
+    function emitStatusbarEvent(evtType) {
+      var evt = new CustomEvent(evtType);
+      Statusbar.handleEvent(evt);
```

Comment:
is there any reason not to `window.dispatchEvent(evt)` instead? (it's more real)